app/controllers/application_controller.rb

```diff
@@ -1851,10 +1856,30 @@ def identify_tl_or_perf_record
     identify_record(params[:id], controller_to_model)
   end
 
-  def assert_privileges(*features)
-    if features.none? { |feature| role_allows?(:feature => feature) }
-      raise MiqException::RbacPrivilegeException, _('The user is not authorized for this task or item.')
-    end
+  def assert_privileges(*features, any: false)
+    msg = "Features checked: #{features.join(', ')}"
+
+    pass =
+      if any
+        role_allows?(:feature => features.first, :any => true)
+      else
+        features.any? { |feature| role_allows?(:feature => feature) }
+      end
+
+    log_privileges(pass, msg)
+    raise MiqException::RbacPrivilegeException, _('The user is not authorized for this task or item.') unless pass
+  end
+
+  def log_privileges(pass, details = nil)
+    # This is called with or without a current user and possibly a fake request such as in test.
+    username    = current_userid rescue nil
+    role_name   = current_user.miq_user_role.name rescue nil
+    http_method = request.respond_to?(:request_method) ? request.request_method : nil
+    path        = request.respond_to?(:filtered_path)  ? request.filtered_path  : nil
```

Comment:
If the `request` is a double such as in test, we can't call a method it's not expecting so instead, we only treat it like a `request` if it's acts like a `request`.